App.js

```diff
@@ -1,112 +1,38 @@
-/**
- * Sample React Native App
- * https://github.com/facebook/react-native
- *
- * @format
- * @flow strict-local
- */
+import React, {useState, useEffect} from 'react';
+import {View, Text} from 'react-native';
+import NetInfo from '@react-native-community/netinfo';
+import Snackbar from 'react-native-snackbar';
+
+const App = () => {
+  const [connectionType, setConnectionType] = useState('');
+  const [isConnected, setIsConnected] = useState(null);
+  const unsubscribe = NetInfo.addEventListener(state => {
```

Comment:
Нужно отписаться от прослушивания события, при unmount. Таким образом ты подписываешься on mount и отписываешься on unmount компонента в одном useEffect. а саму логику можно вынести в отдельный useCallback
```suggestion
      useEffect(() => {
        const unsubscribe = NetInfo.addEventListener(onConnectionChange);
   
        return () => {
            unsubscribe();
        };
    }, []);
```